### DISCUSSION TITLE
Make Page Scrollable

### DISCUSSION THREAD

**Discussion by ahmedsaheed** [charmbracelet/bubbletea#387] [Q&A]
Hi, 

I am trying to make a view scrollable but i wans't able to archieve this.

Here is a [quick lookup](https://github.com/ahmedsaheed/commander/blob/09e98472d33ca876cded257ae1ebb9db991c35c6/main.go#L202) of the code. Could you please point out what i might be doing wrong and also a correctw way to implement this.

Thanks a million

**Comment by bashbunni** [comment]
Hey @ahmedsaheed sorry for the delayed response!

Looks like you're not updating the `viewport` at any point. I would recommend doing a `tea.Batch` so you can send whatever messages should be processed by the `viewport` (like up, down, j, k, etc) after updating the other components or when the `viewport` is focused.

You can see in [this example](https://github.com/charmbracelet/bubbletea/blob/master/examples/pager/main.go#L98) that the `viewport` is getting updated with each call to `Update`. Let me know if you still have questions, I'd be happy to dive in further.
